modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/FormulaObjectFieldBusinessType.java

```diff
@@ -0,0 +1,142 @@
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.SetUtil;
+import com.liferay.portal.vulcan.extension.PropertyDefinition;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marcela Cunha
+ */
+@Component(
+	immediate = true,
+	property = "object.field.business.type.key=" + ObjectFieldConstants.BUSINESS_TYPE_FORMULA,
+	service = {
+		FormulaObjectFieldBusinessType.class, ObjectFieldBusinessType.class
+	}
+)
+public class FormulaObjectFieldBusinessType implements ObjectFieldBusinessType {
+
+	@Override
+	public Set<String> getAllowedObjectFieldSettingsNames() {
+		return SetUtil.fromArray("output", "script");
+	}
+
+	@Override
+	public String getDBType() {
+		return ObjectFieldConstants.DB_TYPE_STRING;
+	}
+
+	@Override
+	public String getDDMFormFieldTypeName() {
+		return DDMFormFieldTypeConstants.TEXT;
+	}
+
+	@Override
+	public String getDescription(Locale locale) {
+		return _language.get(
+			ResourceBundleUtil.getModuleAndPortalResourceBundle(
+				locale, getClass()),
+			"algorithm-that-derives-its-value-from-other-fields");
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(
+			ResourceBundleUtil.getModuleAndPortalResourceBundle(
+				locale, getClass()),
+			"formula");
+	}
+
+	@Override
+	public String getName() {
+		return ObjectFieldConstants.BUSINESS_TYPE_FORMULA;
+	}
+
+	@Override
+	public Map<String, Object> getProperties(
+			ObjectField objectField,
+			ObjectFieldRenderingContext objectFieldRenderingContext)
+		throws PortalException {
+
+		Map<String, Object> properties = new HashMap<>();
+
+		ListUtil.isNotEmptyForEach(
+			_objectFieldSettingLocalService.getObjectFieldObjectFieldSettings(
+				objectField.getObjectFieldId()),
+			objectFieldSetting -> properties.put(
+				objectFieldSetting.getName(), objectFieldSetting.getValue()));
+
+		return properties;
+	}
+
+	@Override
+	public PropertyDefinition.PropertyType getPropertyType() {
+		return PropertyDefinition.PropertyType.TEXT;
+	}
+
+	@Override
+	public void validateObjectFieldSettings(
+			long objectDefinitionId, String objectFieldName,
+			List<ObjectFieldSetting> objectFieldSettings)
+		throws PortalException {
+
+		ObjectFieldBusinessType.super.validateObjectFieldSettings(
+			objectDefinitionId, objectFieldName, objectFieldSettings);
+
+		Map<String, String> objectFieldSettingsValues = new HashMap<>();
+
+		objectFieldSettings.forEach(
+			objectFieldSetting -> objectFieldSettingsValues.put(
+				objectFieldSetting.getName(), objectFieldSetting.getValue()));
+
+		_ddmExpressionFactory.createExpression(
```

Comment:
You are not treating if it fails. If you want an example, you could check ObjectValidationRuleLocalServiceImpl.java > _validateScript();
Let me know what you think!

Reply:
Hi @marcelabc I'm treating it [here](https://github.com/liferay-objects/liferay-portal/pull/1217) in another pull request.

Reply:
makes sense. thx